x-pack/legacy/plugins/maps/public/embeddable/map_embeddable_factory.ts

```diff
@@ -136,39 +127,23 @@ export class MapEmbeddableFactory extends EmbeddableFactory {
     }
 
     return embeddable;
-  }
+  };
 
-  async createFromState(
-    state: { title?: string; layerList?: unknown[] },
-    input: MapEmbeddableInput,
-    parent: IContainer,
-    renderTooltipContent: RenderToolTipContent,
-    eventHandlers: EventHandlers
-  ) {
-    const layerList = state && state.layerList ? state.layerList : getInitialLayers();
+  create = async (input: MapEmbeddableInput, parent?: IContainer) => {
```

Comment:
See comment above. I think the separation of `input` and `state` are important and would like to see them remain separate. If you don't like the idea of `state` in this method, then how about passing `MapEmbeddableConfig` or setting `layerList` with a setter.

Reply:
I can set layerList with a setter.